src/fallible.rs

```diff
@@ -90,9 +90,8 @@ pub fn devicename_os() -> Result<OsString> {
 
 /// Get the host device's hostname.
 ///
-/// Limited to a-z, A-Z, 0-9, and dashes.  This limit also applies to
-/// [`devicename()`] when targeting Windows.  Usually hostnames are
-/// case-insensitive, but it's not a hard requirement.
+/// Usually hostnames are case-insensitive, but it's
+/// not a hard requirement.
```

Comment:
Same documentation on platform-specific character limitations will be needed here too

```suggestion
/// Usually hostnames are case-insensitive, but it's
/// not a hard requirement.
///
/// FIXME: Document platform-specific character limitations
```